From bee76a184e164af67b2566ddcda092bea7fd6bb7 Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Tue, 15 Sep 2026 20:06:12 +0200
Subject: [PATCH] BUILD: Allow the sysconfdir to be overridden

CMAKE_INSTALL_SYSCONFDIR was pinned to the absolute path /etc, so
`make install` wrote outside CMAKE_INSTALL_PREFIX and a relocatable or
unprivileged install was not possible. EOS_INSTALL_SYSCONFDIR now
selects it and still defaults to /etc.
---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c052d6e9a..326a671ce 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -98,7 +98,11 @@ if (NOT PACKAGEONLY)
   include(EosCompileFlags)
 endif()
 
-set(CMAKE_INSTALL_SYSCONFDIR /etc)
+if (NOT EOS_INSTALL_SYSCONFDIR)
+  set(EOS_INSTALL_SYSCONFDIR /etc)
+endif()
+
+set(CMAKE_INSTALL_SYSCONFDIR ${EOS_INSTALL_SYSCONFDIR})
 include(EosFindLibs)
 
 include(CTest)
